tests/bases/test_collections.py

```diff
@@ -278,3 +277,87 @@ def compute(self):
     mc2 = MetricCollection([MyAccuracy(), DummyMetric()])
     mc(torch.tensor([0, 1]), torch.tensor([0, 1]), kwarg="kwarg")
     mc2(torch.tensor([0, 1]), torch.tensor([0, 1]), kwarg="kwarg", kwarg2="kwarg2")
+
+
+@pytest.mark.parametrize(
+    "metrics, expected",
+    [
+        # single metric forms its own compute group
+        (Accuracy(3), {0: ["Accuracy"]}),
+        # two metrics of same class forms a compute group
+        ({"acc0": Accuracy(3), "acc1": Accuracy(3)}, {0: ["acc0", "acc1"]}),
+        # two metrics from registry froms a compute group
+        ([Precision(3), Recall(3)], {0: ["Precision", "Recall"]}),
+        # two metrics from different classes gives two compute groups
+        ([ConfusionMatrix(3), Recall(3)], {0: ["ConfusionMatrix"], 1: ["Recall"]}),
+        # multi group multi metric
+        (
+            [ConfusionMatrix(3), CohenKappa(3), Recall(3), Precision(3)],
+            {0: ["ConfusionMatrix", "CohenKappa"], 1: ["Recall", "Precision"]},
+        ),
+        # Complex example
+        (
+            {
+                "acc": Accuracy(3),
+                "acc2": Accuracy(3),
+                "acc3": Accuracy(num_classes=3, average="macro"),
+                "f1": F1Score(3),
+                "recall": Recall(3),
+                "confmat": ConfusionMatrix(3),
+            },
+            {0: ["acc", "acc2", "f1", "recall"], 1: ["acc3"], 2: ["confmat"]},
+        ),
+    ],
+)
+def test_check_compute_groups(metrics, expected):
+    """Check that compute groups are formed after initialization."""
+    m = MetricCollection(deepcopy(metrics), compute_groups=True)
+    # Construct without for comparison
+    m2 = MetricCollection(deepcopy(metrics), compute_groups=False)
+
+    assert len(m.compute_groups) == len(m)
+    assert m2.compute_groups == {}
+
+    preds = torch.randn(10, 3).softmax(dim=-1)
+    target = torch.randint(3, (10,))
+    m.update(preds, target)
+    m2.update(preds, target)
+
+    assert m.compute_groups == expected
+    assert m2.compute_groups == {}
+
+    preds = torch.randn(10, 3).softmax(dim=-1)
+    target = torch.randint(3, (10,))
+    # compute groups should kick in here
+    m.update(preds, target)
+    m2.update(preds, target)
+
+    # compare results for correctness
+    res_cg = m.compute()
+    res_without_cg = m2.compute()
+    for key in res_cg.keys():
+        assert torch.allclose(res_cg[key], res_without_cg[key])
```

Comment:
shall we include some time measure to be sure that the update does not explode...
so for example one test with 3x the same classic metric and reasonable nb batches to be sure that without > with